core/slim_sim.cpp

```diff
@@ -3815,6 +3815,9 @@ void SLiMSim::SimplifyTreeSequence(void)
 		}
 	}
 	
+	// our tables copy needs to have a population table to be able to sort it
+	WritePopulationTable(&tables);
```

Comment:
This is a change: tskit now checks that each `node.population` refers to a valid entry in the `population` table.